src/hooks/useSarcophagusNegotiation.ts

```diff
@@ -138,7 +147,12 @@ export function useSarcophagusNegotiation() {
         await createEncryptionKeypair();
         const { encryptedShards, encryptedShardsTxId } = await uploadAndSetEncryptedShards();
         console.log('initiating sarco negotiation');
-        await initiateSarcophagusNegotiation(encryptedShards, encryptedShardsTxId);
+        const archaeologistSignatures = await initiateSarcophagusNegotiation(encryptedShards, encryptedShardsTxId);
+        console.log(
+          'arch sigs:',
+          archaeologistSignatures
+        );
+        await uploadAndSetDoubleEncryptedFile();
```

Comment:
Good lawd yes this is so much more digestible than the borderline spaghetti flow we currently have going on.

If there's no significant drawbacks to Promisifying all this then I'm 200% all in. We could go ahead and remove as much "flow control" flags as possible, and simply await function responses 🔥